stl/inc/xtr1common

```diff
@@ -175,6 +175,13 @@ template <class _Ty, class... _Types>
 _INLINE_VAR constexpr bool _Is_any_of_v = // true if and only if _Ty is in _Types
     disjunction_v<is_same<_Ty, _Types>...>;
 
+#if _HAS_CXX20
+// FUNCTION is_constant_evaluated
+_NODISCARD constexpr bool is_constant_evaluated() noexcept {
+    return __builtin_is_constant_evaluated();

```

Comment:
Keeping `is_constant_evaluated` in its header would be more helpful in enforcing to include `<type_traits>` for it in user's code. STL code could just use `__builtin_is_constant_evaluated` directly.

Reply:
That's true, but (1) most STL headers already include `<type_traits>`, so users can already be extremely lazy about including it (the number of headers that this change actually affects is relatively small), and (2) variously using the public function or the builtin makes searching harder. We could always use the builtin, but historically, using compiler builtins outside of their usual contexts was a great way to trigger compiler bugs (this was definitely the case for some of the type traits builtins). `__builtin_is_constant_evaluated` isn't known to be affected by such issues, but I remain wary.

We generally haven't tried to be super disciplined about ensuring that only `<meow>` provides `std::meow`; that's certainly the default, but we've lifted things into centralized headers whenever we've needed them widely. The throughput concern is minimal for this microscopic function; the argument for cleanliness here would be portability between implementations, I suppose.

In the long run, I hope modules will make these issues moot.

Reply:
We'll go ahead and merge this PR as-is, but please file an issue if you'd like us to reconsider the conventions here (it is certainly not a clear choice). :smile_cat: